server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1302,6 +1305,17 @@ public Host maintain(final PrepareForMaintenanceCmd cmd) {
             throw new CloudRuntimeException("Host is already in state " + host.getResourceState() + ". Cannot recall for maintenance until resolved.");
         }
 
+        if (SET_HOST_DOWN_TO_MAINTENANCE.value() && (host.getStatus() == Status.Down)) {
```

Comment:
@ravening consider the value in zone, as the config scope is specified as Zone.

```suggestion
        if (SET_HOST_DOWN_TO_MAINTENANCE.valueIn(host.getDataCenterId()) && (host.getStatus() == Status.Down)) {
```

Reply:
> @ravening consider the value in zone, as the config scope is specified as Zone.
> 
> 
> 
> ```suggestion
> 
>         if (SET_HOST_DOWN_TO_MAINTENANCE.valueIn(host.getDataCenterId()) && (host.getStatus() == Status.Down)) {
> 
> ```

@sureshanaparti i don't think we need that because the host can exist only in one zone at any time but I can change the setting to zone scope instead of global scope

Reply:
@ravening Currently, the config scope is Zone, do you want to mark that as Global scope?

Reply:
@sureshanaparti I dont think I need to make changes here as config is at zone level..setting in one zone doesnt mess up setting in other zones

Reply:
@ravening May be some confusion here. My point is, use the config value specified at the scope defined in the config parameter, else update the config scope accordingly.

The config "SET_HOST_DOWN_TO_MAINTENANCE" defined above uses Zone Scope. So, In order to pick the Zone specific value of that config, you have to use "_valueIn(<zoneId>)_", not _value()_ (this picks the global config value). If your use case is to set the config at Cloud level, then change Scope to Global.

Reply:
This observation makes sense. Can you please take a look at _org.apache.cloudstack.framework.config.ConfigKey.valueIn()_ @ravening?